src/nvim/api/vim.c

```diff
@@ -1565,11 +1565,14 @@ ArrayOf(Dictionary) nvim_get_keymap(String mode)
 ///               or "!" for |:map!|, or empty string for |:map|.
 /// @param  lhs   Left-hand-side |{lhs}| of the mapping.
 /// @param  rhs   Right-hand-side |{rhs}| of the mapping.
+///               When called from Lua, also accepts a Lua function
+///               to call when the mapping is executed.
+
 /// @param  opts  Optional parameters map. Accepts all |:map-arguments|
 ///               as keys excluding |<buffer>| but including |noremap|.
 ///               Values are Booleans. Unknown key is an error.
 /// @param[out]   err   Error details, if any.
-void nvim_set_keymap(String mode, String lhs, String rhs, Dict(keymap) *opts, Error *err)
+void nvim_set_keymap(String mode, String lhs, Object rhs, Dict(keymap) *opts, Error *err)
```

Comment:
Type of existing param cannot change in a stable API method, we'll need to use `opts` in this case.

Reply:
Isn't this still backward compatible ?